### PR TITLE
Integrate jitsi-meet in OpenSlides

### PR DESCRIPTION
**Pull request by tsiegleauq**
- minimal jitsi client in the bottom right of the screen
- can be shown and hidden like a messenger
- allows to mute, unmute, call, stop call
- automatically connects to a conference
- shows a list of users connected to the room
- jitsi iframe is currently hidden
- "open in jitsi meet" link
- only one connection will be opened if using multiple tabs
- JITSI_DOMAIN and JITSI_ROOM_NAME must be present in the settings.py

### CODE REVIEW

**Comment by tsiegleauq** [comment]
I played a little more with jitsi's high level library. Although customization is limited, functionality can surely be wrapped inside an adapter. We should offer the full featured Jitsi-client nevertheless.

A jitsi window can now be shown and hidden anytime, can move around, stays in focus even on navigation and even works nicely in the background.
Opening OpenSlides automatically connects to a room defined by the "event name" on the meet.openslides.com server. Clients are usually muted and on audio only mode.
The connected user will have their short_name on jitsi. Hanging up will detach the jitsi client rather than jumping to the select-room screen.

Worked pretty fine for me so far. Please test around.

**Comment by emanuelschuetze** [comment]
I can't test your branch because I get a 504 (Gateway timeout) with a fresh db after restart server:

e.g. `Error occured while trying to proxy to: localhost:4200/apps/users/login/`
Maybe some files are missing?

**Comment by tsiegleauq** [comment]
@emanuelschuetze cannot reproduce. The previous commit had some proxy files changed. I suppose you are on the previous commit. Compare the commit hash.

**Comment by emanuelschuetze** [comment]
Works now correcly with new commit. Thx.

**Comment by tsiegleauq** [comment]
@emanuelschuetze updated:
reloading / reconnecting is more stable
Jitsi is only active in the first tab
link to external jitsi meet client
cleaned to toolbar in the jitsi client
server configs - default should work. Changes on the fly should also work.
mobile clients should work

**Comment by tsiegleauq** [comment]
new:

- transpatent mute button
- call button is now in accent
- call button will be present as long as the user was not joining a room (this will prevent some errors)
- current user is in list
- projector is now compatible with rtc functions
- focus current speaker with bold text
- user list is scrollable (does neither shrink now grow)
- firefox is allowed

**Comment by tsiegleauq** [comment]
new:
- Rooms will be secured with passwords if "JITSI_ROOM_PASSWORD" was set in settings.py
- rename events reflect in OpenSlides

**Comment by tsiegleauq** [comment]
new:
- hide audio conference button if enable-jitsi was not set to true
- show the current speaker above user list

**Comment by FinnStutzenstein** [comment]
Hi, I pushed the first version with jwt support.

Settings.py:
```
JITSI_DOMAIN = "localhost:8443"
JITSI_ROOM_NAME = "test"
# The issuer is also the app_id
JITSI_ISSUER = "os3_server"
JITSI_AUDIENCE = "os3_server"
JITSI_SECRET = "my_secret"
```

docker-jitsi-meet .env
```
# Enable authentication
ENABLE_AUTH=1

# Enable guest access
ENABLE_GUESTS=1

# Select authentication type: internal, jwt or ldap
AUTH_TYPE=jwt

# JWT authentication
#

# Application identifier
JWT_APP_ID=os3_server

# Application secret known only to your token
JWT_APP_SECRET=my_secret

# (Optional) Set asap_accepted_issuers as a comma separated list
JWT_ACCEPTED_ISSUERTS=os3_server

# (Optional) Set asap_accepted_audiences as a comma separated list
JWT_ACCEPTED_AUDIENCES=os3_server
```

After every change in `.env` you have to delete the `~/.jitsi-meet-cfg` (or what ever you have set up for the `CONFIG` variable) and recreate the config folders as given in the readme.

I did not test `ENABLE_GUESTS=0` yet. But if you are a guest and new in the room, there is a popup that you have to wait for a organisator to join (via jwt). I still haven't found out, if you can specify the role in the jwt. Also it is still possible to change one's name, if the iframe is visible, so the name is not locked...

**Comment by tsiegleauq** [comment]
> I still haven't found out, if you can specify the role in the jwt

I found this repository.
https://github.com/nvonahsen/jitsi-token-moderation-plugin

In the Jitsi forum, I found the following by one of the developers ( https://community.jitsi.org/t/jwt-tokens-and-guest-access/18119/2 )

> How we handle that is that we enable all_owners module so everyone in the room is moderator, enable jwt and enable enableUserRolesBasedOnToken. The later make sure that guests(those with no token) cannot do stuff like calling, recording etc.

That was closest thing I found to a clear statement to actual role based authentication.
So, if the user does have a JWT they will be moderators. Without a JWT users will be guests. I suppose we can either include the LUA module mentioned above or adjust the JWT path in OpenSlides accordingly

To sum it up:
If the user is in a certain OpenSlides group, send JWT, the user will be a Jitsi-moderator.
if not, do not send a JWT, the user will log in as a guest in Jtisi and thus not being a moderator.

A totally different approach:
We block direct Jitsi access, thus only allow OpenSlides-iFrame access, make everyone a Jitsi moderator and block moderating features in the Jitsi-iFrame / the OpenSlides UI depending on our own user roles.

**Comment by tsiegleauq** [comment]
Since this supposed to be merged without JWT, I am going to remove the last commit. Please create a branch of this branch to work on it. Send you the patch file of your commit via email.

File: openslides/core/config_variables.py [hunk 1]
```diff
@@ -122,6 +122,24 @@ def get_config_variables():
         subgroup="System",
     )
 
+    yield ConfigVariable(
+        name="general_system_show_audio_conference",
```
Comment (FinnStutzenstein) [inline openslides/core/config_variables.py:126]: Please rename to `general_system_conference_show`

File: openslides/core/config_variables.py [hunk 2]
```diff
@@ -122,6 +122,24 @@ def get_config_variables():
...
+        default_value=False,
+        input_type="boolean",
+        label="Show the audio conferece integration at the bottom right corner",
+        weight=142,
+        subgroup="System",
+    )
+
+    yield ConfigVariable(
+        name="general_system_auto_connect_conference",
```
Comment (FinnStutzenstein) [inline openslides/core/config_variables.py:135]: Please rename to `general_system_conference_auto_connect`

**Comment by tsiegleauq** [comment]
all done 
image
